refactor(test): reorganize OpenAI test file structure

CatherineSue opened the pull request:
<!-- Thank you for your contribution! We appreciate it. The following guidelines will help improve your pull request and facilitate feedback. If anything is unclear, don't hesitate to submit your pull request and ask the maintainers for assistance. -->

## Motivation

This PR aims to reorganize the OAI test suites for better tests management.

Thanks to @key4ng's suggestion
<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Modifications

- Introduce a hierarchical directory structure under `test/srt/openai` to group tests by functionality:
  - `basic/`: For fundamental OpenAI API compatibility tests (completions, chat, etc.)
  - `features/`: For advanced functionality (JSON mode, hidden states, EBNF, etc.)
  - `function_call/`: For function calling and tool choice tests
  - `validation/`: For input validation and request limit tests
  - `embeddings/`: For embedding-specific server tests

- Merged `test_embedding_openai_server.py` and `TestOpenAIServerEmbedding` in `test_openai_server.py` into one
- Move all relevant OpenAI test files from `test/srt/` and the old `test/srt/openai_server/` into their new respective directories.

- Update all test file paths in `test/srt/run_suite.py` to match the new structure, ensuring `per-commit` and `per-commit-amd` suites are correct.

- Add previously unreferenced tests (`test_openai_server_ebnf.py`, `test_openai_server_ignore_eos.py`) to the test suites.


<!-- Describe the changes made in this PR. -->

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

CatherineSue committed:
refactor(test): reorganize OpenAI test file structure

- Introduce a hierarchical directory structure under `test/srt/openai` to group tests by functionality:
  - `basic/`: For fundamental OpenAI API compatibility tests (completions, chat, etc.)
  - `features/`: For advanced functionality (JSON mode, hidden states, EBNF, etc.)
  - `function_call/`: For function calling and tool choice tests
  - `validation/`: For input validation and request limit tests
  - `embeddings/`: For embedding-specific server tests

- Merged `test_embedding_openai_server.py` and `TestOpenAIServerEmbedding` in `test_openai_server.py` into one
- Move all relevant OpenAI test files from `test/srt/` and the old `test/srt/openai/` into their new respective directories.

- Update all test file paths in `test/srt/run_suite.py` to match the new structure, ensuring `per-commit` and `per-commit-amd` suites are correct.

- Add previously unreferenced tests (`test_openai_server_ebnf.py`, `test_openai_server_ignore_eos.py`) to the test suites.

CatherineSue committed:
Rename `test_openai_adapter.py` to `test_jinja_template_utils.py`

CatherineSue committed:
Rename `test/srt/openai` to `test/srt/openai_server` and update test command at the top of each file

CatherineSue committed:
Fix lint

JustinTong0323 commented:
https://github.com/sgl-project/sglang/blob/72676cd6c0eead68286cf9c8df5c67f580ff477b/test/srt/test_cache_report.py#L93-L106

Change L100 to     `assert first_cached_tokens <= self.min_cached` could resolve the Assertion error. I test locally.

CatherineSue committed:
Fix test_cache_report

- self.min_cached should be a conservative upper bound of how many tokens to be cached before any user's true text, for instance, self.min_cached could include system prompt in chat template
- first_cached_tokens can be equal or smaller than this bound. It doesn't have to be strictly small.

Thanks to @JustinTong0323 and @jhinpan to fix this issue.
